test/indexer/test_tdr.py

```diff
@@ -236,9 +244,21 @@ def _drs_file_id(self, file_id):
 class TestPlugin(tdr.Plugin):
     tinyquery: tinyquery.TinyQuery
 
+    def _set_utc_tzinfo(self, d: Optional[datetime]) -> Optional[datetime]:
+        return None if d is None else d.replace(tzinfo=timezone.utc)
+
     def _run_sql(self, query: str) -> BigQueryRows:
         columns = self.tinyquery.evaluate_query(query).columns
         num_rows = one(set(map(lambda c: len(c.values), columns.values())))
+        # Tinyquery returns naive datetime objects from a TIMESTAMP type column,
+        # so we manually set the tzinfo back to UTC on these values.
+        # https://github.com/Khan/tinyquery/blob/9382b18b/tinyquery/runtime.py#L215
+        for key, column in columns.items():
+            if column.type == 'TIMESTAMP':
+                values = [self._set_utc_tzinfo(d) for d in column.values]
```

Comment:
If you went with nested function, you could map that function over column.values.

Then again, I think inlining the function into the comprehension would be less circuitous without being overly verbose.

Generally, functions are used with map, expressions with comprehensions. The combination of comprehension with function sticks out.